### PR TITLE
Document that literals with any suffixes are valid as tokens

### PR DESCRIPTION
**Pull request by petrochenkov**
Closes https://github.com/rust-lang/rust/issues/60494

### CODE REVIEW

File: src/tokens.md [hunk 1]
```diff
@@ -88,6 +88,20 @@ evaluated (primarily) at compile time.
 
 #### Suffixes
 
+Any kind of literal (string, integer, etc) with any suffix is valid as a token,
```
Comment (Centril) [inline src/tokens.md]: The suffix must be an IDENT, right?

e.g. `blackhole!("string"4);` is not OK even tho `4` is a suffix of `"string"`

Comment (petrochenkov) [reply src/tokens.md]: >The suffix must be an IDENT, right?

Yes, added a commit with the definition.

Comment (petrochenkov) [reply src/tokens.md]: `4` is not a suffix, `"string"4` is lexed into two literal tokens.

File: src/tokens.md [hunk 2]
```diff
@@ -88,6 +88,23 @@ evaluated (primarily) at compile time.
 
 #### Suffixes
 
+Suffix is a non-raw identifier immediately (without whitespace)
```
Comment (Centril) [inline src/tokens.md]:
```suggestion
A suffix is a non-raw identifier immediately (without whitespace)
```

**Comment by ehuss** [comment]
That section isn't really intended to be the definition, but is rather just a list of examples.  Perhaps it should be reorganized since others have been a bit confused by it.

I would think there also needs to be some updates to the `Lexer` definitions to cover this.  It's unfortunate that this makes essentially two sets of literals, but perhaps just a consolidated section could be created to cover "macro tokens".

I can work on rearranging this some time soonish, but I wouldn't consider this resolved.

**Comment by Centril** [comment]
@ehuss Could you file an issue meanwhile?

**Comment by petrochenkov** [comment]
Yeah, a new reference issue, not reopening that lang issue.
I wasn't too happy about the structure of `src/tokens.md` either, but I also wasn't ready for doing a larger rewrite.

**Pull request merged**